smalltalksrc/VMMaker/VMStructType.class.st

```diff
@@ -222,6 +222,25 @@ VMStructType class >> instVarNamesAndTypesForTranslationDo: aBinaryBlock [
 	self subclassResponsibility
 ]
 
+{ #category : #translation }
+VMStructType class >> instVarTypeDeclarationsDo: aBinaryBlock [
+
+	| instVarNameTypeDeclarations |
+	instVarNameTypeDeclarations := Dictionary new.
+
+	self instVarNamesAndTypesForTranslationDo: [ :ivn :type |
+		instVarNameTypeDeclarations at: ivn put: type ].
+
+
+	self filteredInstVarNames do: [ :ivn |
+		instVarNameTypeDeclarations at: ivn ifAbsent: [
+			TranslationError signal:
+				'Missing type declartions for instance variable ' , ivn ] ].
+
+	self filteredInstVarNames do: [ :ivn |
+		aBinaryBlock value: ivn value: (instVarNameTypeDeclarations at: ivn) ]
```

Comment:
These two `do:`s can be merged... I don't know what is better.

Reply:
I think it makes more sense to merge them